File: CS/ClientServer_Wcf_StandardAuth.Module/BusinessObjects/TestBO.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;

namespace ClientServer_Wcf_StandardAuth.Module.BusinessObjects {
    [DefaultClassOptions]
    public class TestBO : XPObject {
        public TestBO(Session session) : base(session) { }
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue<string>("Name", value); }
        }
    }

    [DefaultClassOptions]
    public class FullAccessBO : XPObject {
        public FullAccessBO(Session session) : base(session) { }
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue<string>("Name", value); }
        }
    }

    [DefaultClassOptions]
    public class InaccessibleByUserBO : XPObject {
        public InaccessibleByUserBO(Session session) : base(session) { }
        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue<string>("Name", value); }
        }
    }
}

File: VB/ClientServer_Wcf_StandardAuth.Module/BusinessObjects/TestBO.vb
Imports System.Text
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base

Namespace ClientServer_Wcf_StandardAuth.Module.BusinessObjects
    <DefaultClassOptions> _
    Public Class TestBO
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Property Name() As String
            Get
                Return GetPropertyValue(Of String)("Name")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Name", value)
            End Set
        End Property
    End Class

    <DefaultClassOptions> _
    Public Class FullAccessBO
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Property Name() As String
            Get
                Return GetPropertyValue(Of String)("Name")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Name", value)
            End Set
        End Property
    End Class

    <DefaultClassOptions> _
    Public Class InaccessibleByUserBO
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Property Name() As String
            Get
                Return GetPropertyValue(Of String)("Name")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Name", value)
            End Set
        End Property
    End Class
End Namespace
